api/views.py

```diff
@@ -557,3 +578,48 @@ def transfer_students(request):
     src.students.remove(student)
         
     return JsonResponse({'transfers': transfers})
+
+
+@user_passes_test(is_teacher)
+def semesters(request):
+    semesters = Semester.objects.all()
+
+    semesters_json = serializers.serialize('json', semesters)
+
+    # TODO: When upgrading, see: https://docs.djangoproject.com/en/4.2/ref/request-response/#httpresponse-objects
+    # for a way to set content type.
+    return HttpResponse(semesters_json,
+                        content_type="application/json")
+
+
+
+
+@user_passes_test(is_teacher)
+def import_activities(request):
```

Comment:
`@require_http_methods(["POST"])`

Reply:
Oh, I didn't know that decorator. I found there's also `@require_POST()`, so I'll use this one directly.